federator/controllers/utils.go

```diff
@@ -76,68 +75,165 @@ func IsObjAMKOClusterType(ctx context.Context, name string) bool {
 	return true
 }
 
-func FederateObjects(ctx context.Context, memberClusters []KubeContextDetails, objList []client.Object) error {
+func DeleteObjsOnAllMemberClusters(ctx context.Context, memberClusters []KubeContextDetails, namespace string, obj client.Object) error {
 	for _, m := range memberClusters {
-		for _, obj := range objList {
-			clusterClient := *m.client
-			// decide whether to create or update
-			var gcObj gslbalphav1.GSLBConfig
-			var gdpObj gdpalphav2.GlobalDeploymentPolicy
-			var err error
-
-			// identify the type of object and put in the relevant variable
-			if obj.GetObjectKind().GroupVersionKind() == gcGVK {
-				err = clusterClient.Get(ctx, types.NamespacedName{
-					Namespace: obj.GetNamespace(),
-					Name:      obj.GetName(),
-				}, &gcObj)
-			} else if obj.GetObjectKind().GroupVersionKind() == gdpGVK {
-				err = clusterClient.Get(ctx, types.NamespacedName{
-					Namespace: obj.GetNamespace(),
-					Name:      obj.GetName(),
-				}, &gdpObj)
+		clusterClient := *m.client
+		if err := clusterClient.DeleteAllOf(ctx, obj, &client.DeleteAllOfOptions{
+			ListOptions: client.ListOptions{
+				Namespace: namespace,
+			},
+		}); err != nil {
+			return fmt.Errorf("error in deleting all objects of kind %s in %s namespace and in %s cluster: %v",
```

Comment:
why is this all or nothing? For example, if m clusters exist and n is a subset of m. n has lost connectivity or are permanently down. Looks like the m-n clusters will also not get update in the process?

Reply:
Fair point. This was done for simplicity of implementation and retries. But, you are right, this can cause issues if any clusters is/are permanently down. Let me re-work on this.